### PR TITLE
Add PROP_ATOM_FLOATING for conveyor belt immunity

### PR DESCRIPTION
**Pull request by stonepillars**
[cleanliness]

## About the PR 

Add PROP_ATOM_FLOATING which makes you immune to conveyor belts. This could be expanded in the future (ex. walking straight over railings or pits)

## Why's this needed? 

So flockdrones can use it in #8711

### CODE REVIEW

File: code/mob/dead.dm [hunk 1]
```diff
@@ -8,6 +8,7 @@
 /mob/dead/New()
 	..()
 	src.flags |= UNCRUSHABLE
+	APPLY_ATOM_PROPERTY(src, PROP_ATOM_FLOATING, src)
```
Comment (amylizzle) [inline code/mob/dead.dm:11]: should probably apply to `/mob/living/intangible` also

Comment (stonepillars) [reply code/mob/dead.dm:11]: the old one didn't check! but will see if that causes problems

**Comment by FlameArrow57** [comment]
What about for /obj/critters with flying set to TRUE? There are some cases in the code I think where flying prevents them from falling into holes/lava

**Comment by stonepillars** [comment]
Does not remove the old check for flying obj/critters. This property is equal but not for critters. Maybe I'll get around to adding them to critters but that's a major refactor of code that should be removed and replaced with something else anyway.

**Review by frawhst — APPROVED** [review]

**Pull request merged**